flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBulkEmail.cls

```diff
@@ -0,0 +1,19 @@
+public inherited sharing class SendBulkEmail {
```

Comment:
Mass email is only restricted to contacts users and leads!

Let’s say I have a custom object that has a email field Mass email won’t work. The api of mass email has setTargetId method that accepts only lead contact and user IDs.

The idea behind bulk email is we provide a interface that can accept a user defined apex class. admins create multiple requests input using user defined class and pass in the list to send emails!

There is a sample flow I have added that has a working example how to work with the bulk send email!

Reply:
@alexed1 In the past, we've been working one-on-one with users who have true bulk email issues, basically using apex to create a collection of sendBetterEmail requests - and that worked (at least before the bulkification, which I have not tested).  However, it is WAY too complex for most developers, not to mention admins.  When Mohith and I discussed this a few weeks ago, the concept of adding a flow-based creation process for the request collections was broached.  HOWEVER, this is for <1% of the cases and comes with the cost of making it more complex for the standard user.

Can you give me a couple of days to look at this and mull it over?  He makes a good point about using the standard singleEmail and massEmail classes (and the limitations thereof, e.g., limited TargetIds and WhatIds for massEmail).

My issue is now with the complexity.  I can no longer (easily) follow the code since the mapping was introduced (at a savings up to 4 dml queries per request),  so even seeing what it is currently doing stretches the limits of my capabilities.

Reply:
Further Note:  The last few comments/support requests  on sendBetterEmail have been related to ineligible Object types in both the TargetId and the RecordId/What Id fields and the restrictions thereof which correlate to limitations of Messaging.Email with the two Messaging.MassEmailMessage and Messaging.SingleEmailMessage classes